packages/earth-map/src/sagas/layers/index.ts

```diff
@@ -26,10 +26,70 @@ import { DATA_INDEX_QUERY } from '../model';
 import { IIndex } from 'modules/indexes/model';
 import { ILayer } from 'modules/layers/model';
 import { IWidget } from 'modules/widget/model';
+import { setSidebarPanel } from 'modules/sidebar/actions';
+import { EPanels } from 'modules/sidebar/model';
 import { setWidgets, setWidgetsLoading, setWidgetsError } from 'modules/widgets/actions';
 import { setIndexesList } from 'modules/indexes/actions';
-import { setLayersList } from 'modules/layers/actions';
-import { getGroup } from 'sagas/saga-utils';
+import {
+  setLayersList,
+  setLayersSearch,
+  setLayersLoading,
+  setLayersSearchResults,
+  setLayersSearchAvailableFilters,
+  resetLayersResults,
+  nextLayersPage,
+} from 'modules/layers/actions';
+import { getGroup, getLayers, onlyMatch } from 'sagas/saga-utils';
+import { serializeFilters } from 'utils/filters';
+import { fetchLayers } from 'services/layers';
+import { LAYER_QUERY } from '../model';
+
+export default function* layers() {
+  // @ts-ignore
+  yield takeLatest(onlyMatch(setSidebarPanel, EPanels.LAYERS), searchLayers);
+  yield takeLatest(setLayersSearch, searchLayers);
+  yield takeLatest(nextLayersPage, nextPage)
+}
+
+function* searchLayers(params) {
+  yield put(resetLayersResults());
+  const { meta } = yield nextPage(params);
+  yield put(setLayersSearchAvailableFilters(meta.filters));
+}
+
+/**
+ * Used to retrieve a page of results
+ */
+export function* nextPage({ payload }) {
+  const group = yield select(getGroup);
+  const { search } = yield select(getLayers);
+  const { filters, search: userInput } = search;
+  const { pageCursor, pageSize } = payload;
+
+  const filterQuery = serializeFilters(filters);
+
+  yield put(setLayersLoading(true));
+  const options = {
+    ...LAYER_QUERY,
+    ...(!!userInput && { search: userInput }),
+    ...(!!filterQuery && { filter: filterQuery }),
+    'page[cursor]': pageCursor ? pageCursor : -1,
+    ...(pageSize && { page: { size: pageSize } }),
+    ...{ group: group.toString() },
+  };
+  const page = yield call(fetchLayers, options);
+  const { data: results, meta } = page;
+
+  yield put(
+    setLayersSearchResults({
+      results,
+      nextPageCursor: meta.pagination.nextCursor,
+    })
+  );
+  yield put(setLayersLoading(false));
+
+  return page;
+}
 
 export function* preloadLayers({ payload }) {
   const group = yield select(getGroup);
```

Comment:
in preloadLayers there is still a call for layers from dashboards, is that still used?